ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -169,6 +170,18 @@ public void setIngestIfNotExists(ArrayList<String> ingestIfNotExists) {
             fullAdditionalProperties.put("ingestIfNotExists", ingestIfNotExistsJson);
         }
         fullAdditionalProperties.putAll(additionalProperties);
+
+        switch (ingestionMapping.IngestionMappingKind) {
```

Comment:
Those key names are part of the ingestion properties object we pass the DM, I think it is belongs in here because its a part of the ingestion properties definition solely and not as logic in the ingestion mapping object

Reply:
Not sure I'm following the logic here. Let's discuss...